File: Задание3/prog3.kt
fun numReverse(num: Int): Int {
    var length = 0
    var numCopy = num
    var digit = 10

    while (numCopy != 0) {
        if (numCopy % 10 >= digit) {
            return 0
        }

        length++
        digit = numCopy % 10
        numCopy /= 10
    }

    var numReversed = 0
    var originalNum = num

    for (i in 0 until length) {
        numReversed = numReversed * 10 + originalNum % 10
        originalNum /= 10
    }

    return if (length > 1) numReversed else 0
}

fun main() {
    val count = readLine()!!.toInt()

    var remainingCount = count

    while (remainingCount > 0) {
        val number = readLine()!!.toInt()
        val result = numReverse(number)

        if (result != 0) {
            print("$result ")
        }

        remainingCount--
    }
}

File: Задание3/prog3.swift
func numReverse(_ num: Int) -> Int {
    var length = 0
    var numCopy = num
    var digit = 10
    
    while numCopy != 0 {
        if numCopy % 10 >= digit {
            return 0
        }
        
        length += 1
        digit = numCopy % 10
        numCopy /= 10
    }
    
    var numReversed = 0
    var numCopy2 = num
    
    for _ in 0..<length {
        numReversed = numReversed * 10 + numCopy2 % 10
        numCopy2 /= 10
    }
    
    return length > 1 ? numReversed : 0
}

var count = Int(readLine()!)!

while count > 0 {
    let number = Int(readLine()!)!
    let result = numReverse(number)
    
    if result != 0 {
        print(result, terminator: " ")
    }
    
    count -= 1
}